From 096455ea0f9f007f134f042bbe53fef23f1f1d10 Mon Sep 17 00:00:00 2001
From: Victor Gomes <victorgomes@chromium.org>
Date: Wed, 26 Jan 2022 15:08:25 +0100
Subject: [PATCH] [gm] Fuchsia support

Adds the archs fuchsia_x64 and fuchsia_arm64.

No-Try: true
Change-Id: I5e12a436fc206fecc910965e9597b7e05be23e26
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/3416247
Auto-Submit: Victor Gomes <victorgomes@chromium.org>
Reviewed-by: Jakob Kummerow <jkummerow@chromium.org>
Commit-Queue: Jakob Kummerow <jkummerow@chromium.org>
Cr-Commit-Position: refs/heads/main@{#78779}
---
 tools/dev/gm.py | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/tools/dev/gm.py b/tools/dev/gm.py
index 613065d5b11..42ff0b5e7c9 100755
--- a/tools/dev/gm.py
+++ b/tools/dev/gm.py
@@ -43,7 +43,8 @@ BUILD_TARGETS_ALL = ["all"]
 
 # All arches that this script understands.
 ARCHES = ["ia32", "x64", "arm", "arm64", "mipsel", "mips64el", "ppc", "ppc64",
-          "riscv64", "s390", "s390x", "android_arm", "android_arm64", "loong64"]
+          "riscv64", "s390", "s390x", "android_arm", "android_arm64", "loong64",
+          "fuchsia_x64", "fuchsia_arm64"]
 # Arches that get built/run when you don't specify any.
 DEFAULT_ARCHES = ["ia32", "x64", "arm", "arm64"]
 # Modes that this script understands.
@@ -291,7 +292,7 @@ class Config(object):
     cpu = "x86"
     if self.arch == "android_arm":
       cpu = "arm"
-    elif self.arch == "android_arm64":
+    elif self.arch == "android_arm64" or self.arch == "fuchsia_arm64":
       cpu = "arm64"
     elif self.arch == "arm64" and _GetMachine() in ("aarch64", "arm64"):
       # arm64 build host:
@@ -310,7 +311,7 @@ class Config(object):
   def GetV8TargetCpu(self):
     if self.arch == "android_arm":
       v8_cpu = "arm"
-    elif self.arch == "android_arm64":
+    elif self.arch == "android_arm64" or self.arch == "fuchsia_arm64":
       v8_cpu = "arm64"
     elif self.arch in ("arm", "arm64", "mipsel", "mips64el", "ppc", "ppc64",
                        "riscv64", "s390", "s390x", "loong64"):
@@ -322,6 +323,8 @@ class Config(object):
   def GetTargetOS(self):
     if self.arch in ("android_arm", "android_arm64"):
       return ["target_os = \"android\""]
+    elif self.arch in ("fuchsia_x64", "fuchsia_arm64"):
+      return ["target_os = \"fuchsia\""]
     return []
 
   def GetSpecialCompiler(self):
-- 
2.35.1

